airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -251,13 +264,64 @@ public ConnectionRead webBackendUpdateConnection(WebBackendConnectionUpdate webB
     return connectionRead;
   }
 
+  private List<UUID> createOperations(WebBackendConnectionCreate webBackendConnectionCreate)
+      throws JsonValidationException, ConfigNotFoundException, IOException {
+    final List<UUID> operationIds = new ArrayList<>();
+    for (var operationCreate : webBackendConnectionCreate.getWithOperations()) {
+      operationIds.add(operationsHandler.createOperation(operationCreate).getOperationId());
+    }
+    return operationIds;
+  }
+
+  private List<UUID> updateOperations(WebBackendConnectionUpdate webBackendConnectionUpdate)
+      throws JsonValidationException, ConfigNotFoundException, IOException {
+    final ConnectionRead connectionRead =
+        connectionsHandler.getConnection(new ConnectionIdRequestBody().connectionId(webBackendConnectionUpdate.getConnectionId()));
+    final List<UUID> originalOperationIds = new ArrayList<>(connectionRead.getOperationIds());
+    final List<UUID> operationIds = new ArrayList<>();
+    for (var operationUpdate : webBackendConnectionUpdate.getWithOperations()) {
+      if (!originalOperationIds.contains(operationUpdate.getOperationId())) {
```

Comment:
you're trying to handle the case where the are new operations added during the update, right? would it make sense to just check if they have ids or not? the new ones i assume don't have ids yet. 

also (assuming the new ones don't have ids) don't actually be valid `OperationUpdate` objects, i don't think. I think you might need to add an extra struct that's `OperationCreateOrUpdate` where the id is optional. I think that would make things a bit clearer. The way it is now is a little hard to follow and it means that the required fields in the structs aren't respected. there's probably a better idea than what i suggested, but definitely whatever you pick it would nice for it to be a little clearer.